### PR TITLE
Issues/#2 cpu mining

### PR DESCRIPTION
**Pull request by macterra**
Added SHA-3 hashing to CPU mining (configure with `--enable-keccak --enable-cpumining`)

### CODE REVIEW

**Comment by macterra** [comment]
TBD change the git submodule to link to Equibit/libblkmaker instead of bitcoin/libblkmaker...

File: malgo/keccak.c [hunk 1]
```diff
@@ -84,7 +86,11 @@ void keccak1(unsigned char *out, const unsigned char *inraw, unsigned inrawlen)
 	UINT64 Esa, Ese, Esi, Eso, Esu;
 	
 	memcpy(temp, inraw, inrawlen);
-	temp[inrawlen++] = 1;
+#ifdef SHA3
+    temp[inrawlen++] = 0x6;
```
Comment (r-soltani) [inline malgo/keccak.c]: what is the 0x6 value ?

Comment (macterra) [reply malgo/keccak.c]: The difference between Keccak and SHA-3...
> The padding change is the only difference, this allows future tree hashing modes as well as the current SHAKE outputs to generate different digests given the same security parameters and message inputs. Up to 4 additional bits are added, which keeps the full padding inside a byte boundary, making implementations with octet only input able to switch to SHA-3 from Keccak with change to only a single line of code.

from https://crypto.stackexchange.com/a/15728

File: miner.c [hunk 2]
```diff
@@ -13258,7 +13279,7 @@ int main(int argc, char *argv[])
 	atexit(bfg_atexit);
 
 	b58_sha256_impl = my_blkmaker_sha256_callback;
-	blkmk_sha256_impl = my_blkmaker_sha256_callback;
+    blkmk_sha256_impl = my_blkmaker_sha3_callback;
```
Comment (r-soltani) [inline miner.c]: Does this change still allow the miner to switch to SHA256 through command line

Comment (macterra) [reply miner.c]: No

**Pull request merged**